lib/ansible/modules/cloud/vmware/vmware_get_info.py

```diff
@@ -0,0 +1,164 @@
+    - Valid filters for folder object type - folders, parent_folders, names, datacenters, type.
+    - Valid filters for host object type - folders, hosts, names, datacenters, clusters, connection_states.
+    - Valid filters for network object type - folders, types, names, datacenters, networks.
+    - Valid filters for resource_pool object type - resource_pools, parent_resource_pools, names, datacenters, hosts, clusters.
+    - Valid filters for virtual_machine object type - folders, resource_pools, power_states, vms, names, datacenters, hosts, clusters.
+    - content_library, local_library, subscribed_library, content_type, tag, category does not take any filters.
+    default: []
+    type: list
+'''
+
+EXAMPLES = r'''
+- name: Get Datacenter object information
+  vmware_get_info:
+    object_type: datacenter
+    filters:
+      - names: Asia-Datacenter1
+  register: datacenter_result
+
+- set_fact:
+    datacenter_obj: "{{ datacenter_result.object_info.value[0]['datacenter'] }}"
+
+- name: Get all clusters from Asia-Datacenter1
+  vmware_get_info:
+    object_type: cluster
+    filters:
+      - datacenters: "{{ datacenter_obj }}"
+  register: clusters_result
+'''
+
+RETURN = r'''
+object_info:
+    description: facts about the given VMware object
+    returned: always
+    type: dict
+    sample: {
+        "value": [
+            {
+                "cluster": "domain-c42",
+                "drs_enabled": false,
+                "ha_enabled": false,
+                "name": "Asia-Cluster1"
+            }
+        ]
+    }
+'''
+
+from ansible.module_utils.connection import Connection
+from ansible.module_utils.basic import AnsibleModule
+
+
+def main():
+    argument_spec = dict(
+        object_type=dict(type='str', default='datacenter'),
+        filters=dict(type='list', default=[]),
+    )
+
+    module = AnsibleModule(argument_spec=argument_spec,
+                           supports_check_mode=True)
+
+    conn = Connection(module._socket_path)
+
+    rest_result = {}
+    object_type = module.params['object_type']
+    object_type_url_map = dict(
+        datacenter=('/vcenter/datacenter', ['folders', 'datacenters', 'names']),
+        cluster=('/vcenter/cluster', ['folders', 'datacenters', 'names', 'clusters']),
+        datastore=('/vcenter/datastore', ['folders', 'datacenters', 'names', 'datastores', 'types']),
+        folder=('/vcenter/folder', ['folders', 'datacenters', 'names', 'parent_folders', 'type']),
+        host=('/vcenter/host', ['folders', 'datacenters', 'names', 'hosts', 'clusters', 'connection_states']),
+        network=('/vcenter/network', ['folders', 'datacenters', 'names', 'types', 'networks']),
+        resource_pool=('/vcenter/resource-pool', ['datacenters', 'names', 'hosts', 'parent_resource_pools', 'clusters', 'resource_pools']),
+        virtual_machine=('/vcenter/vm', ['folders', 'datacenters', 'names', 'hosts', 'power_states', 'clusters', 'resource_pools', 'vms']),
```

Comment:
Could we use `vm` too, maybe as an alias. This is shorter, clear enough IMO and it's actually already VMware naming convention.